### PR TITLE
Add direct-csi plugin

### PR DESCRIPTION
**Pull request by nitisht**

### CODE REVIEW

File: kubectl-direct-csi/util/util.go [hunk 1]
```diff
@@ -0,0 +1,750 @@
...
+	rbacv1 "k8s.io/api/rbac/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/minio/direct-csi/kubectl-direct-csi/util/randomstring"
+	"github.com/minio/direct-csi/pkg/clientset"
+)
+
+func retry(f func() error) error {
```
Comment (wlan0) [inline kubectl-direct-csi/util/util.go:38]: we can remove this function

File: kubectl-direct-csi/util/util.go [hunk 2]
```diff
@@ -0,0 +1,750 @@
...
+		return err
+	})
+}
+
+func CreateStorageClass(ctx context.Context, kClient clientset.Interface, name string) error {
+	allowExpansion := false
+	bindingMode := storagev1.VolumeBindingImmediate
+	allowedTopologies := []corev1.TopologySelectorTerm{}
+	parameters := map[string]string{
+		"disable.csi.storage.k8s.io/provisioner-secret-name":      "direct-csi-min-io",
```
Comment (wlan0) [inline kubectl-direct-csi/util/util.go:114]: These parameters are not parsed anymore. We can remove this

File: kubectl-direct-csi/util/util.go [hunk 3]
```diff
@@ -0,0 +1,750 @@
...
+			Name: name,
+			Labels: map[string]string{
+				createByLabel: directCSIController,
+			},
+			Annotations: map[string]string{
+				"rbac.authorization.kubernetes.io/autoupdate": "true",
+			},
+		},
+		[]rbacv1.PolicyRule{
+			rbacv1.PolicyRule{
```
Comment (wlan0) [inline kubectl-direct-csi/util/util.go:236]: There is a new RBAC role I added to create CRDs. Please update this to add that RBAC role

**Review by wlan0 — CHANGES_REQUESTED** [review]

File: kubectl-direct-csi/util/util.go [hunk 4]
```diff
@@ -0,0 +1,750 @@
...
+	// }, f); err != nil {
+	// 	if !errors.IsAlreadyExists(err) {
+	// 		glog.Errorf("CSIDriver creation failed: %v", err)
+	// 		return err
+	// 	}
+	// }
+	return nil
+}
+
+func CreateCSIDriver(ctx context.Context, kClient clientset.Interface, name string) error {
```
Comment (Praveenrajmani) [inline kubectl-direct-csi/util/util.go:53]: Question: Shouldn't we be calling these functions on `kubectl direct-csi install`?

Comment (nitisht) [reply kubectl-direct-csi/util/util.go:164]: It was not hooked, done now.

File: kubectl-direct-csi/util/util.go (hunk 1 above)
Comment (nitisht) [reply kubectl-direct-csi/util/util.go:99]: Updated the method, IMO it is useful, any reason you want to remove?

**Comment by nitisht** [comment]
Ready for review now, PTAL @wlan0 @Praveenrajmani

File: kubectl-direct-csi/util/util.go (hunk 1 above)
Comment (wlan0) [reply kubectl-direct-csi/util/util.go:102]: There is a retry method available already in client-go/tools/retry

**Review by wlan0 — APPROVED** [review]

File: kubectl-direct-csi/util/util.go (hunk 1 above)
Comment (nitisht) [reply kubectl-direct-csi/util/util.go:102]: I see, let me remove this in next PR then.

**Pull request merged**